src/darwin/Framework/CHIP/ServerEndpoint/MTRServerAttribute.h

```diff
@@ -0,0 +1,65 @@
+/**
+ *    Copyright (c) 2024 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#import <Foundation/Foundation.h>
+#import <Matter/MTRBaseClusters.h>
+#import <Matter/MTRDefines.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+/**
+ * A representation of an attribute implemented on a server cluster by an
+ * MTRDeviceController.  An attribute has an identifier and a value, and may or
+ * may not be writable.
+ */
+MTR_NEWLY_AVAILABLE
+@interface MTRServerAttribute : NSObject <NSCopying>
```

Comment:
What does it really mean to copy this type, now that it has internal state? Should it be copyable?

Reply:
Well, it copies the current value, attribute ID, etc.  It does not copy the "I am attached to something" state, since the new thing is not thus attached....

I could take out the NSCopying here (and then also make clusters and endpoints not copyable).

Reply:
If it's a "dumb object" (no moving parts), then copying makes sense.

But as soon as it becomes a "live handle" to something (such that calling methods causes side effects, like setting attribute values and sending updates to subscribers), then it stops feeling like it should be copyable.

I don't really like API that "sometimes is valid to call" (eg, before you hand it off to the controller). 

Also, I'd expect that if I copy ` b = [a copy]` that `[b isEqualTo: a]` is true, which I'm not sure it really ever could be (given the internal state not copied).

I'm not convinced copying is good if this is a handle.

Reply:
> I don't really like API that "sometimes is valid to call" (eg, before you hand it off to the controller).

The only way to avoid that here is to make all the things that can't change after handing off to a controller init arguments.  Mostly this means the cluster list for the endpoint and the attribute list for the cluster.

@ksperling-apple you asked for the mutator methods for those; are you OK with just making them init arguments?

Reply:
I think removing NSCopying sounds like the right thing to me here.

Reply:
OK, ditching the NSCopying and isEqual bits on MTRServerEndpoint, MTRServerCluster, MTRServerAttribute.